### PR TITLE
Dataframe: schema cache and partitions num

### PR DESCRIPTION
**Pull request by ntviet18**
- read partitions 
- schema cache

### CODE REVIEW

**Comment by ntviet18** [comment]
@fe2s , could you please review?

File: src/test/scala/com/redislabs/provider/redis/rdd/RedisSqlStandaloneSuite.scala [hunk 1]
```diff
@@ -124,6 +124,20 @@ class RedisSqlStandaloneSuite extends FunSuite with ENV with BeforeAndAfterAll w
     }
   }
 
+  test("repartition read/write") {
+    // generate random table, so we can run test multiple times and not append/overwrite data
```
Comment (fe2s) [inline src/test/scala/com/redislabs/provider/redis/rdd/RedisSqlStandaloneSuite.scala]: I think we don't need this comment (it's in the generateTableName function)

**Comment by fe2s** [comment]
@ntviet18 please address the minor comment, other than that it looks good

**Comment by ntviet18** [comment]
@fe2s , thank you, updated

**Pull request merged**